lib/libvarnish/vus.c

```diff
@@ -95,7 +95,12 @@ int
 VUS_bind(const struct sockaddr_un *uds, const char **errp)
 {
 	int sd, e;
-	socklen_t sl = sizeof(*uds);
+	socklen_t sl;
+
+	if (*uds->sun_path)
+		sl = sizeof(*uds);
+	else
+		sl = strchr(uds->sun_path + 1, '\0') - (const char*)uds;
```

Comment:
We could add assertions following from `sun_init()`: `assert(uds->sun_path[1] != '\0');`.
and assert that `strchr()` returns non-null for the same reason.
sl should be checked to not exceed `sizeof(*uds)`.
And all this probably qualifies to wrap it in a function for reuse.